WalletWasabi.Gui/Tabs/WalletManager/LoadWalletViewModel.cs

```diff
@@ -440,67 +394,56 @@ public async Task<KeyManager> LoadKeyManagerAsync(bool requirePassword, bool isH
 				var walletName = selectedWallet.WalletName;
 				if (isHardwareWallet)
 				{
+					var client = new HwiClient(Global.Network);
+
 					if (selectedWallet.HardwareWalletInfo is null)
 					{
 						SetValidationMessage("No hardware wallets detected.");
 						return null;
 					}
 
-					if (!selectedWallet.HardwareWalletInfo.Initialized)
+					if (!selectedWallet.HardwareWalletInfo.IsInitialized())
 					{
-						IEnumerable<HardwareWalletInfo> hwis;
 						try
 						{
 							IsHardwareBusy = true;
 							MainWindowViewModel.Instance.StatusBar.TryAddStatus(StatusBarStatus.SettingUpHardwareWallet);
-							if (!await HwiProcessManager.SetupAsync(selectedWallet.HardwareWalletInfo))
+
+							// Setup may take a while for users to write down stuff.
+							using (var ctsSetup = new CancellationTokenSource(TimeSpan.FromMinutes(21)))
 							{
-								throw new Exception("Setup failed.");
+								// Trezor T doesn't require interactive mode.
+								if (selectedWallet.HardwareWalletInfo.Model == HardwareWalletModels.Trezor_T
+								|| selectedWallet.HardwareWalletInfo.Model == HardwareWalletModels.Trezor_T_Simulator)
+								{
+									await client.SetupAsync(selectedWallet.HardwareWalletInfo.Model, selectedWallet.HardwareWalletInfo.Path, false, ctsSetup.Token);
+								}
+								else
+								{
+									await client.SetupAsync(selectedWallet.HardwareWalletInfo.Model, selectedWallet.HardwareWalletInfo.Path, true, ctsSetup.Token);
+								}
 							}
 
+							// Only Windows waits for the process to exit.
```

Comment:
```suggestion
```
Maybe this comment should be removed because the following code was removed
```
if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
{
	MainWindowViewModel.Instance.StatusBar.TryAddStatus(StatusBarStatus.ConnectingToHardwareWallet);
	await EnumerateHardwareWalletsAsync();
}
```

Reply:
Removed, thanks.